### ISSUE TITLE
Front Matter Description is not used for metadata but summary 

### ISSUE THREAD

**Issue by schachem** [nunocoracao/blowfish#690]
**Describe the bug**
I noticed that the Theme by default tries to use the front matter parameter "summary" to set the metadata description.
If there is none (that was the case for me, do not write summaries manually and have set auto summary length to 0), it uses the overall site description as metadata description.

While I see no inherent problem with that per se, I must make the remark that the theme documentation to the best of my knowledge does not mention that. 
In fact, it tells you the exact opposite, namely that as I would have expected that the front matter description parameter is used for setting the metadata description (which it does not).

I am not sure how to best remedy that or what was originally intended?
I am almost assuming that this was a typo of some sort?
Even when using the summary feature, I would rather have the option to set the metadata description separately.

So I guess, the easiest and maybe also intenteded behavior is to just use the description and not summary of an page for the metadata?

**To Reproduce**
Visit any blowfish.page article site.  
Inspect the metadata and notice that the site description is used as a fallback because no summary exists.

**Expected behavior**
I would expect the description parameter to be used for setting metadata.
Using the site description as fallback seems fine.

**Comment by schachem** [comment]
To make a long story short, I propose to change:

https://github.com/nunocoracao/blowfish/blob/92f26f9bdd20345767cfe23fa519f12b1c5aaece/layouts/partials/head.html#L17

to:

` {{ with .Params.description | default .Site.Params.description -}} `

This then would adhere to the documentation in my opinion.
Otherwise, I would propose to change the docs :)

Thanks for the awesome theme!

**Comment by schachem** [comment]
If you want me do, I can make a proper pull request this time :D

**Comment by nunocoracao** [comment]
Happy to accept a PR but not with this solution. Let's keep the same logic as of today to avoid breaking changes to existing users. What I mean is:

1) yes, update docs to whatever the final solution is
2) keep using the summary in order not to break previous installations
3) also include description that way it doesn't really matter which one users pick it will get used in the right places 


Thoughs?

**Comment by schachem** [comment]
Based on your requirements I see two potential solutions.

First:
Try to set metadata description as:  `.Params.description > .Params.Summary > .Site.Params.description` 
This relies on the assumption that all people using the summary did not explicitly set a description for the article.
I would argue that should be the case, as I do not see a reason to set the description parameter when using the summary other than wanting that to be the metadescription, because it is not used for anything at the moment I guess.

Second:
The other (very safe) option would be to set the metadata description as:
`.Params.Summary > .Params.description > .Site.Params.description` 

However I feel that does not make a lot of sense for the scenario where I explicitly want to set a different metadescription than the summary. Because in my case, I might want to start using the summaries in the future as they are a nice thing visually on the blog, however I want to be able to set a distinct metadescription.

Third option:
Create an global switch in the params allowing the user to opt for either the first or second option; and set the second option as default.

Please let me know what should be implemented, and I'll try to do it :)

**Comment by nunocoracao** [comment]
Sorry, I'm vetoing the third one - not because I don't like it - because the theme is getting out of hand with so many possible config combinations, and it's getting hard to test/maintain. 

Between 1 and 2 I prefer 2 as that will break less stuff in the wild. I do agree with your argument for having different descriptions. 

I would go with 
- Blowfish will not support having a different summary and description for now
- the theme will always rely on the summary for both fields OR the description if available OR the site one 

Thoughts?

**Comment by schachem** [comment]
Yes I can understand why option 3 is not preferred.
I am not sure if I understand correctly:

Basically we should do option two, correct?
That would at least allow users that do not use summaries but descriptions to end up with a more meaningful meta description.

**Comment by nunocoracao** [comment]
Yes. Option 2. I prefer making sure current users don't have to deal with breaking changes as that is a big pain when it happens.

**Comment by ilmax** [comment]
@nunocoracao I would like to challenge the choice here, I think option 1 was the better one because the description value is significant in SEO practices so I'd like the ability to control it precisely, without losing the summary feature. 

Ideally, I'd like the description to be set using the frontmatter to be the picked one, then the summary, and finally the site description.

What do you think?
